src/impression.js

```diff
@@ -62,10 +62,18 @@ export function maybeTrackImpression(win) {
         dev().warn('IMPRESSION', error);
       });
 
-  Services.viewerForDoc(win.document).isTrustedViewer().then(isTrusted => {
+  const viewer = Services.viewerForDoc(win.document);
+  const isTrustedViewerPromise = viewer.isTrustedViewer();
+  const isTrustedReferrerPromise = viewer.isTrustedReferrer();
+  Promise.all([
```

Comment:
Just realize, we don't need to wait for all promise to be resolved. As long as we get one of them to be true we're good to go. This is minor and we don't need to worry about it here : ) most because these two promises will resolve at approx same time.

Reply:
cool, I will keep it as it is then